WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardReaderSupportDeterminer.swift

```diff
@@ -59,7 +60,16 @@ final class CardReaderSupportDeterminer: CardReaderSupportDetermining {
 
     @MainActor
     func deviceSupportsLocalMobileReader() async -> Bool {
-        await withCheckedContinuation { continuation in
+        /// There may be crashes due to multiple consecutive calls checkDeviceSupport
+        /// Local mobile reader (Tap to Pay) support shouldn't change during the app lifecycle
+        /// Only a change in operating system version can affect it which would require a restart of the app
```

Comment:
Are we absolutely sure about this?

What you write makes sense, but to me it seems this is a black box of Apple's and Stripe's code, and caching the result for the lifetime of the app adds a small risk of caching an incorrect result, perhaps when the app has just launched.

I'm probably being over cautious, but what about caching it for 30s only, to prevent repeated calls? Would that be enough, from your investigations?

Reply:
I'm not absolutely sure. And you're right being cautious. I was thinking about timing the cache as well. I think it's reasonable to limit it to 30s.